skcosmo/sample_selection/_voronoi_fps.py

```diff
@@ -0,0 +1,307 @@
+    calculations outweighs the increase from book-keeping. For simple
+    metrics (such as Euclidean distance), VoronoiFPS will likely not
+    accelerate, and may decelerate, computations when compared
+    to FPS.
+
+    Parameters
+    ----------
+
+    n_trial_calculation: integer, default=4
+        Number of calculations used for the switching point between Voronoi FPS
+        and traditional FPS (for detail look at full_fraction).
+
+    full_fraction: float, default=None
+        Proportion of calculated distances from the total number of features at
+        which the switch from Voronoi FPS to FPS occurs.
+        At a certain number of distances to be calculated,
+        the use of Voronoi FPS becomes unreasonably expensive due to the associated
+        costs connected with reading data from the memory. The switching point
+        depends on many conditions, and it is determined "in situ" for optimal
+        use of the algorithm. Determination is done with a few test calculations
+        and memory operations.
+    """
+
+    def __init__(
+        self, n_trial_calculation=4, full_fraction=None, initialize=0, **kwargs
+    ):
+
+        self.n_trial_calculation = n_trial_calculation
+        self.full_fraction = full_fraction
+        self.initialize = initialize
+        super().__init__(selection_type="sample", **kwargs)
+
+    def score(self, X=None, y=None):
+        """
+        Returns the Haussdorf distances of all samples to previous selections
+
+        NOTE: This function does not compute the importance score each time it
+        is called, in order to avoid unnecessary computations. The haussdorf
+        distance is updated in :py:func:`self._update_post_selection`
+
+        Parameters
+        ----------
+        X : ignored
+        y : ignored
+
+        Returns
+        -------
+        haussdorf : Haussdorf distances
+        """
+        return self.haussdorf_
+
+    def get_distance(self):
+        """
+
+        Traditional FPS employs a column-wise Euclidean
+        distance for feature selection, which can be expressed using the covariance matrix
+        :math:`\\mathbf{C} = \\mathbf{X} ^ T \\mathbf{X}`
+
+        .. math::
+            \\operatorname{d}_c(i, j) = C_{ii} - 2 C_{ij} + C_{jj}.
+
+        For sample selection, this is a row-wise Euclidean distance, which can
+        be expressed in terms of the Gram matrix :math:`\\mathbf{K} = \\mathbf{X} \\mathbf{X} ^ T`
+
+        .. math::
+            \\operatorname{d}_r(i, j) = K_{ii} - 2 K_{ij} + K_{jj}.
+
+        Returns
+        -------
+
+        haussdorf : ndarray of shape (`n_to_select_from_`)
+                     the minimum distance from each point to the set of selected
+                     points. once a point is selected, the distance is not updated;
+                     the final list will reflect the distances when selected.
+
+        """
+        return self.haussdorf_
+
+    def get_select_distance(self):
```

Comment:
Reading the docstring, I'm not sure what's the difference between get_distance and this function is. It sounds like both give the haussdorf distance of a point when it was selected.